frame/dvm/evm/precompiles/bridge/bsc/src/lib.rs

```diff
@@ -63,85 +49,26 @@ where
 		is_static: bool,
 	) -> PrecompileResult {
 		let mut helper = PrecompileHelper::<T>::new(input, target_gas);
-		let (selector, data) = helper.split_input()?;
+		let (selector, _) = helper.split_input()?;
 		let action = Action::from_u32(selector)?;
 
 		// Check state modifiers
 		helper.check_state_modifier(context, is_static, StateMutability::View)?;
 
 		let output = match action {
-			Action::VerfiySingleStorageProof => {
-				// Storage: BSC FinalizedCheckpoint (r:1 w:0)
-				helper.record_gas(1, 0)?;
-
-				let params = BscSingleStorageVerifyParams::decode(data)
-					.map_err(|_| helper.revert("Decode input failed"))?;
-				let finalized_header = darwinia_bridge_bsc::Pallet::<T>::finalized_checkpoint();
-				let proof = EthereumStorageProof::new(
-					params.lane_address,
-					params.storage_key,
-					params.account_proof,
-					params.storage_proof,
-				);
-				let storage_value = EthereumStorage::<Vec<u8>>::verify_storage_proof(
-					finalized_header.state_root,
-					&proof,
-				)
-				.map_err(|_| helper.revert("Verify proof failed"))?;
-				abi_encode_bytes(storage_value.0.as_slice())
-			},
-			Action::VerifyMultiStorageProof => {
+			Action::StateRoot => {
 				// Storage: BSC FinalizedCheckpoint (r:1 w:0)
 				helper.record_gas(1, 0)?;
 
-				let params = BscMultiStorageVerifyParams::decode(data)
-					.map_err(|_| helper.revert("Decode input failed"))?;
 				let finalized_header = darwinia_bridge_bsc::Pallet::<T>::finalized_checkpoint();
-				let key_size = params.storage_keys.len();
-				if key_size != params.storage_proofs.len() {
-					return Err(helper.revert("Storage keys not match"));
-				}
-				if key_size > MAX_MULTI_STORAGEKEY_SIZE {
-					return Err(helper.revert("Key'size too large"));
-				}
-				let storage_values: Result<Vec<Vec<u8>>, _> = (0..key_size)
-					.map(|idx| {
-						let storage_key = params.storage_keys[idx];
-						let storage_proof = params.storage_proofs[idx].clone();
-						let proof = EthereumStorageProof::new(
-							params.lane_address,
-							storage_key,
-							params.account_proof.clone(),
-							storage_proof,
-						);
-						let storage_value = EthereumStorage::<Vec<u8>>::verify_storage_proof(
-							finalized_header.state_root,
-							&proof,
-						);
-						match storage_value {
-							Ok(value) => {
-								return Ok(value.0.clone());
-							},
-							Err(err) => {
-								// if the key not exist, we should return Zero Value due to the
-								// Zero Value will not be stored.
-								if err == StorageProofError(ProofError::TrieKeyNotExist) {
-									return Ok(vec![]);
-								} else {
-									return Err(helper.revert("Verfiy proof failed"));
-								}
-							},
-						}
-					})
-					.collect();
-				abi_encode_array_bytes(storage_values?)
+				finalized_header.state_root
 			},
 		};
 
 		Ok(PrecompileOutput {
 			exit_status: ExitSucceed::Returned,
 			cost: helper.used_gas(),
-			output,
+			output: output.as_bytes().to_vec(),
```

Comment:
Is an `abi_encode` needed here?

Reply:
I think it's necessary.

Reply:
No need, Cause is state_root is Hash256.